bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/handler/MoonHandler.java

```diff
@@ -64,6 +65,16 @@ public void publishPositionalInfo() {
         Double longitude = thingConfig.longitude;
         moonCalc.setPositionalInfo(Calendar.getInstance(), latitude != null ? latitude : 0,
                 longitude != null ? longitude : 0, moon);
+
+        Eclipse eclipses = moon.getEclipse();
+        eclipses.getKinds().forEach(eclipseKind -> {
+            Calendar eclipseDate = eclipses.getDate(eclipseKind);
+            Position position = getPositionAt(eclipseDate.toInstant().atZone(timeZoneProvider.getTimeZone()));
+            if (position != null) {
+                eclipses.set(eclipseKind, eclipseDate, position);
+            }
+        });
```

Comment:
This logic seems to be used in multiple places, can it be refactored into the Eclipse class instead?